### DISCUSSION TITLE
Mockable plugin instances & methods

### DISCUSSION THREAD

**Discussion by Ehesp** [firebase/flutterfire#6566] [Feature request]
## Bug report

Currently to access the reworked plugins, you access them via statics, e.g:

```
FirebaseFirestore.instance
```

Since they're statics, they are not mockable.

## Solutions

One solution proposed by @rrousselGit is to expose a mock static method for each plugin, e.g:

```
void main() {
  final mock = FirebaseFirestore.mock();

  afterEach(mock.reset);

  test('foo', () {
    mock.mockDocument('key', {'foo': 42});
  });
}
```

**Comment by TahaTesser** [comment]
Just adding the `core` label  to get this issue out of backlog since this is  for all packages, not sure we need to add  all the package labels
